bindings/java/src/main/com/apple/foundationdb/tuple/TupleUtil.java

```diff
@@ -63,6 +64,28 @@
 	private static final byte[] TRUE_ARR           = new byte[]{0x27};
 	private static final byte[] VERSIONSTAMP_ARR   = new byte[]{0x33};
 
+	private static final ThreadLocal<DecodeResult> DECODE_RESULT_THREAD_LOCAL =
+			new ThreadLocal<DecodeResult>() {
+				@Override
+				protected DecodeResult initialValue() {
+					return new DecodeResult();
+				}
+			};
+	private static final ThreadLocal<EncodeResult> ENCODE_RESULT_THREAD_LOCAL =
+			new ThreadLocal<EncodeResult>() {
+				@Override
+				protected EncodeResult initialValue() {
+					return new EncodeResult();
+				}
+			};
+	private static final ThreadLocal<byte[]> NINE_BYTES_THREAD_LOCAL =
```

Comment:
This could probably do with an explanatory comment saying what this is used for just because it might not be immediately obvious to people why there is a thread-local variable that's just a byte array of length 9. I'm also generally against mutable global variables, so I guess I'd like to know how much of a savings this optimization gets us, if you know (though obviously, this will depend on how many integers one is `Tuple`-packing).